### ISSUE TITLE
Missing type="text" on inputs in prerendered HTML-files

### ISSUE THREAD

**Issue by roydigerhund** [preactjs/preact#3224]
**Describe the bug**
On input fields in prerendered HTML-files, the type attribute is missing if it has the default value "text". Strangely this is not happening when testing directly with the "preact-render-to-string" package in this [CodeSandbox example](https://codesandbox.io/s/preact-render-to-string-test-with-inputs-bdis2?file=/src/index.js).

But it happens when I prerender the same content `preact build`.

**To Reproduce**
1. Prerender a file with an input with `type="text"` in it.
2. Check the prerendered index.html file, the type attribute should be gone

**Expected behavior**
The `type="text"` attribute should not be stripped away, even though it is the default value. It might for example be used for CSS styling.

**Comment by marvinhagemeister** [comment]
@roydigerhund The `text` attribute is present for me in the linked codesandbox. I'm not aware of any special behavior we do around that in `preact-render-to-string`.

<img width="771" alt="Screenshot 2021-07-14 at 14 51 04" src="https://user-images.githubusercontent.com/1062408/125624932-7d755414-3df2-46ba-a741-84cf5d955998.png">

**Comment by roydigerhund** [comment]
@marvinhagemeister yes, in the provided example `preact-render-to-string` will output the `type="text"` attribute. It is however not happening if I prerender the same content as in the CodeSandbox example with preact-cli's `build` script from the package.json.

I will create an example Github-Repo for it.

**Comment by JoviDeCroock** [comment]
I think this is an issue meant for https://github.com/preactjs/preact-cli
